app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,56 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: %i(edit, update)
+  before_action :valid_user, only: %i(edit, update)
+  before_action :check_expiration, only: %i(edit update)
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t ".email_password_reset"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t ".email_not_found"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, t(".can't_empty")
+      render :edit
+    elsif @user.update_attributes user_params
+      log_in @user
+      flash[:success] = t ".password_reset"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit password: :password_confirmation
+  end
+
+  def get_user
+    @user = User.find_by email: params[:email]
```

Comment:
What will happen if it don't find the user with params[:email] ? I meant how you can handle this case?